### DISCUSSION TITLE
Animating translation in a visx/geo projection

### DISCUSSION THREAD

**Discussion by coding-tomato** [airbnb/visx#1569] [General]
Hello!

I was looking for a bit of help, hope this is the right place to ask.

I made an interactive map with visx/zoom and visx/geo, it works amazing and as expected, panning zooming etc.

What would be the best way to smoothly animate translation?

Say, that by double clicking a geo feature, I set a new transformMatrix with a new translation position, and it smoothly interpolates to said position.

Is this possible with visx/react-spring? Time is not on my side and I was hoping some guidance would help me bury my head in the right place haha.

Also, I was hoping to achieve a similar goal with scale, say, setting a new scale and it smoothly interpolates into the new scale setting.

**Comment by williaster** [comment]
hey @coding-tomato 👋  thanks for checking out visx, since this is a usage question I'm going to move this to a discussion.

I don't have a ton of time to dig in currently (maybe someone else can!), my thinking is that `react-spring` itself might be a good avenue for animating this. in can interpolate some data structures, but not all.

**Reply by coding-tomato** [reply]
Thank you! Will look into it.
